ext/random/random.c

```diff
@@ -0,0 +1,871 @@
+	php_random_ce_Random_NumberGenerator_XorShift128Plus->create_object = php_random_rng_xorshfit128plus_new;
+	memcpy(&php_random_rng_xorshift128plus_object_handlers, zend_get_std_object_handlers(), sizeof(zend_object_handlers));
+	php_random_rng_xorshift128plus_object_handlers.offset = XtOffsetOf(php_random_rng, std);
+	php_random_rng_xorshift128plus_object_handlers.free_obj = php_random_rng_common_free_obj;
+	php_random_rng_xorshift128plus_object_handlers.clone_obj = php_random_rng_common_clone_obj;
+
+	/* Random\NumberGenerator\MT19937 */
+	php_random_ce_Random_NumberGenerator_MT19937 = register_class_Random_NumberGenerator_MT19937(php_random_ce_Random_NumberGenerator_RandomNumberGenerator);
+	php_random_ce_Random_NumberGenerator_MT19937->create_object = php_random_rng_mt19937_new;
+	memcpy(&php_random_rng_mt19937_object_handlers, zend_get_std_object_handlers(), sizeof(zend_object_handlers));
+	php_random_rng_mt19937_object_handlers.offset = XtOffsetOf(php_random_rng, std);
+	php_random_rng_mt19937_object_handlers.free_obj = php_random_rng_common_free_obj;
+	php_random_rng_mt19937_object_handlers.clone_obj = php_random_rng_common_clone_obj;
+
+	/* Random\NumberGenerator\Secure */
+	php_random_ce_Random_NumberGenerator_Secure = register_class_Random_NumberGenerator_Secure(php_random_ce_Random_NumberGenerator_RandomNumberGenerator);
+	php_random_ce_Random_NumberGenerator_Secure->create_object = php_random_rng_secure_new;
+	php_random_ce_Random_NumberGenerator_Secure->serialize = zend_class_serialize_deny;
+	php_random_ce_Random_NumberGenerator_Secure->unserialize = zend_class_unserialize_deny;
+	memcpy(&php_random_rng_secure_object_handlers, zend_get_std_object_handlers(), sizeof(zend_object_handlers));
+	php_random_rng_secure_object_handlers.offset = XtOffsetOf(php_random_rng, std);
+	php_random_rng_secure_object_handlers.free_obj = php_random_rng_common_free_obj;
+	php_random_rng_secure_object_handlers.clone_obj = NULL;
+
+	/* RandomInterface */
+	php_random_ce_RandomInterface = register_class_RandomInterface();
+
+	/* Random */
+	php_random_ce_Random = register_class_Random(php_random_ce_RandomInterface);
+	php_random_ce_Random->create_object = php_random_new;
+	memcpy(&php_random_php_random_object_handlers, zend_get_std_object_handlers(), sizeof(zend_object_handlers));
+	php_random_php_random_object_handlers.offset = XtOffsetOf(php_random, std);
+	php_random_php_random_object_handlers.free_obj = php_random_free_obj;
+	php_random_php_random_object_handlers.clone_obj = NULL;
+
+	return SUCCESS;
+}
+/* }}} */
+
+/* {{{ PHP_MINFO_FUNCTION */
+PHP_MINFO_FUNCTION(php_random)
+{
+	php_info_print_table_start();
+	php_info_print_table_header(2, "Random support", "enabled");
+	php_info_print_table_end();
+}
+/* }}} */
+
+/* {{{ php_random_module_entry */
+zend_module_entry php_random_module_entry = {
```

Comment:
same here for random_module_entry, both in PHP_MINIT(random) and PHP_MINIT_FUNCTION(random)